atomicapp/providers/lib/kubeshift/kubernetes.py

```diff
@@ -0,0 +1,171 @@
+"""
+ Copyright 2014-2016 Red Hat, Inc.
+
+ This file is part of Atomic App.
+
+ Atomic App is free software: you can redistribute it and/or modify
+ it under the terms of the GNU Lesser General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ Atomic App is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU Lesser General Public License for more details.
+
+ You should have received a copy of the GNU Lesser General Public License
+ along with Atomic App. If not, see <http://www.gnu.org/licenses/>.
+"""
+
+from urlparse import urljoin
+from urllib import urlencode
+from atomicapp.providers.lib.kubeshift.kubebase import KubeBase
+from atomicapp.providers.lib.kubeshift.exceptions import (KubeKubernetesError)
+from atomicapp.constants import LOGGER_DEFAULT
+import logging
+import re
+logger = logging.getLogger(LOGGER_DEFAULT)
+
+
+class KubeKubernetesClient(object):
+
+    def __init__(self, config):
+        '''
+
+        Args:
+            config (obj): Object of the configuration data
+
+        '''
+
+        # Pass in the configuration data (.kube/config object) to the KubeBase
+        self.api = KubeBase(config)
+
+        # Check the API url
+        url = self.api.cluster['server']
+        if not re.match('(?:http|https)://', url):
+            raise KubeKubernetesError("Kubernetes API URL does not include HTTP or HTTPS")
+
+        # Gather what end-points we will be using
+        self.k8s_api = urljoin(url, "api/v1/")
```

Comment:
Thanks, @cdrage 

Is there a timeline in mind, because by restricting to api/v1, the behavior may be a regression for some?

```
ERROR  :: No kind by that name: Deployment
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/atomicapp-0.5.2-py2.7.egg/atomicapp/cli/main.py", line 130, in cli_func_exec
    cli_func(cli_func_args)
  File "/usr/local/lib/python2.7/dist-packages/atomicapp-0.5.2-py2.7.egg/atomicapp/cli/main.py", line 84, in cli_run
    nm.run(**argdict)
  File "/usr/local/lib/python2.7/dist-packages/atomicapp-0.5.2-py2.7.egg/atomicapp/nulecule/main.py", line 320, in run
    nulecule.run(cli_provider, dryrun)
  File "/usr/local/lib/python2.7/dist-packages/atomicapp-0.5.2-py2.7.egg/atomicapp/nulecule/base.py", line 224, in run
    provider.run()
  File "/usr/local/lib/python2.7/dist-packages/atomicapp-0.5.2-py2.7.egg/atomicapp/providers/kubernetes.py", line 268, in run
    self.api.create(artifact, self.namespace)
  File "/usr/local/lib/python2.7/dist-packages/atomicapp-0.5.2-py2.7.egg/atomicapp/providers/lib/kubeshift/client.py", line 49, in create
    self.connection.create(obj, namespace)
  File "/usr/local/lib/python2.7/dist-packages/atomicapp-0.5.2-py2.7.egg/atomicapp/providers/lib/kubeshift/kubernetes.py", line 62, in create
    kind, url = self._generate_kurl(obj, namespace)
  File "/usr/local/lib/python2.7/dist-packages/atomicapp-0.5.2-py2.7.egg/atomicapp/providers/lib/kubeshift/kubernetes.py", line 142, in _generate_kurl
    raise KubeKubernetesError("No kind by that name: %s" % kind)
KubeKubernetesError: No kind by that name: Deployment
```

I was testing kubernetes v1.3.0-alpha.5, and prior to pulling this commit, I was able to create objects belonging to the extensions/v1beta1 API, such as DaemonSet and Deployment.